src/hitchhiker/tree/core.cljc

```diff
@@ -0,0 +1,690 @@
+(ns hitchhiker.tree.core
```

Comment:
Could you leave a copy of this file as `core.clj` while doing this review, so that I can see the diffs? Looks like github doesn't render the move+diff.